File: doc-examples/jdbc-example-java/src/main/java/example/oracle/ManufacturerRepository.java
package example.oracle;

import example.Manufacturer;
import io.micronaut.context.annotation.Requires;
import io.micronaut.data.jdbc.annotation.JdbcRepository;
import io.micronaut.data.model.query.builder.sql.Dialect;
import io.micronaut.data.repository.GenericRepository;

@JdbcRepository(dialect = Dialect.ORACLE)
@Requires(env="oracle")
public interface ManufacturerRepository extends example.ManufacturerRepository {
}

File: doc-examples/jdbc-example-kotlin/src/main/kotlin/example/ManufacturerRepository.kt
package example

import io.micronaut.data.jdbc.annotation.JdbcRepository
import io.micronaut.data.model.query.builder.sql.Dialect
import io.micronaut.data.repository.kotlin.KotlinCrudRepository

@JdbcRepository(dialect = Dialect.H2)
interface ManufacturerRepository : KotlinCrudRepository<Manufacturer, Long> {

    fun save(name: String): Manufacturer
}
